pandas/core/reshape/reshape.py

```diff
@@ -322,9 +322,7 @@ def get_new_columns(self, value_columns: Index | None):
 
             new_codes = [lab.take(propagator) for lab in value_columns.codes]
         else:
-            # error: Incompatible types in assignment (expression has type "List[Any]",
-            # variable has type "FrozenList")
-            new_levels = [  # type: ignore[assignment]
+            new_levels = [
```

Comment:
``value_columns.levels`` has type ``FrozenList``. In general: Mypy complains if we have unused ignore comments

Reply:
ah missed seeing that during my review. Ill close this PR, thanks

Reply:
@kashifkhan you might be able to get rid of the `# type: ignore` comment by annotating `new_levels` where it's defined (line 320)

Reply:
@MarcoGorelli hmm I can do that, but then on line 327 ````new_levels```` can also be  a List of Index. 

My current thought is to annotate ````new_levels```` to be a ````FrozenList```` on line 320 and then have a new variable ````new_levels_list```` as  a ````List[Index]```` on line 327 . Then change the return statement based on which one is populated ...thoughts?

Reply:
You can just use ``new_levels: FrozenList | list[Index]`` I think?

Reply:
or `sequence[Index]` if it doesn't _need_ to be a list?

Reply:
I think a Sequence should be ok, but since we are only using the variable only within the function we should narrow the type down?

Reply:
@phofl went with list since it is only generated within the function & then passed on to the MultiIndex constructor which needs an Iterable.